src/components/TextEmoji.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import _ from 'underscore';
+import Text from './Text';
+import * as EmojiUtils from '../libs/EmojiUtils';
+import * as StyleUtils from '../styles/StyleUtils';
+import stylePropTypes from '../styles/stylePropTypes';
+import styles from '../styles/styles';
+
+const propTypes = {
+    /** The message text to render */
+    children: PropTypes.string.isRequired,
+
+    /** The message text additional style */
+    style: stylePropTypes,
+
+    /** If TextEmoji was used in the menuItem component */
+    isMenuItem: PropTypes.bool,
```

Comment:
I think this prop is perhaps a bit too domain-specific. Instead, I think the props should be:

1. `emojiStyle` – styles to apply to emojis in the text
1. `plainTextStyle` – styles to apply to non-emoji text in the text

Reply:
thanks for the suggestion I would make changes to this